### PR TITLE
Update CheckboxGroup to new form pattern

### PR DESCRIPTION
**Pull request by zecuria**
Added Checkbox group to new form pattern

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,6 +94,7 @@
     "eslint-plugin-testing-library": "3.10.2",
     "husky": "5.2.0",
     "identity-obj-proxy": "3.0.0",
+    "inquirer": "^8.1.0",
```
Comment (strahinjaajvaz) [inline package.json:97]: can we use exact versions like the rest of the packages?

File: package.json [hunk 2]
```diff
@@ -107,6 +108,7 @@
     "rollup-plugin-delete": "2.0.0",
     "rollup-plugin-peer-deps-external": "2.2.4",
     "rollup-plugin-typescript2": "0.30.0",
+    "semver": "^7.3.5",
```
Comment (strahinjaajvaz) [inline package.json:111]: same as above.

File: scripts/bumpversion.mjs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      name: "passes",
+      type: "list",
+      message: `Confirm that the following percy build passes: ${buildUrl}`,
+      choices: ["yes", "no"],
+    },
+  ]);
+
+  if (passes === "no") {
+    console.log(chalk.red("aborting the release"));
+    return;
```
Comment (strahinjaajvaz) [inline scripts/bumpversion.mjs:173]: Wouldn't it be `process.exit(1)`

File: scripts/bumpversion.mjs [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  }
+
+  const tag = await chooseTag();
+
+  if (!tag) {
+    console.log(chalk.red("aborting the release"));
+    return;
```
Comment (strahinjaajvaz) [inline scripts/bumpversion.mjs:180]: same as above.

File: src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      disabled={disabled}
+      label={label}
+      labelId={labelId}
+      auxId={auxId}
+      helpText={helpText}
+      errors={fieldErrors}
+      testId={testId}
+    >
+      <div ref={mergeRefs([checkboxRef, innerRef])}>
+        {/* @ts-ignore */}
```
Comment (strahinjaajvaz) [inline src/components/CheckboxGroup/CheckboxGroup.tsx:91]: whats the purpose of this `@ts-ignore`

File: src/components/CheckboxGroup/types.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+type NodeOption = {
+  key: string;
+  label: React.ReactNode;
+};
+
+export type CheckboxOption =
+  | StringOption
+  | NodeOption
+  | Readonly<StringOption>
+  | Readonly<NodeOption>;
```
Comment (strahinjaajvaz) [inline src/components/CheckboxGroup/types.ts:17]: Whats the difference in having it as read only AND  just the type? Wouldn't one suffice?

File: src/components/CheckboxGroup/types.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+export type CheckboxGroupProps =
+  | InternalCheckboxGroupProps
+  | DeprecatedCheckboxGroupProps;
+
+export const setDeprecatedProps = (
+  props: CheckboxGroupProps,
+  isDeprecated: boolean
+): props is DeprecatedCheckboxGroupProps => {
```
Comment (strahinjaajvaz) [inline src/components/CheckboxGroup/types.ts:66]: is?

File: scripts/bumpversion.mjs (hunk 3 above)
Comment (zecuria) [reply scripts/bumpversion.mjs:173]: I was fine with returning without really throwing exception but I could do that

File: src/components/CheckboxGroup/types.ts (hunk 7 above)
Comment (zecuria) [reply src/components/CheckboxGroup/types.ts:66]: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

File: src/components/CheckboxGroup/CheckboxGroup.tsx (hunk 5 above)
Comment (zecuria) [reply src/components/CheckboxGroup/CheckboxGroup.tsx]: Stack was not a typescript component but have moved it to typescript now

File: src/components/CheckboxGroup/types.ts (hunk 6 above)
Comment (zecuria) [reply src/components/CheckboxGroup/types.ts:17]: Nope one is not assignable to the other. If we only have `readonly` then we force all consumers to use `as const` which is fine for static properties but is very difficult to do for dynamic ones.

File: src/components/CheckboxGroup/CheckboxGroup.tsx (hunk 5 above)
Comment (strahinjaajvaz) [reply src/components/CheckboxGroup/CheckboxGroup.tsx]: ahh :)

File: src/components/CheckboxGroup/index.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import React, { forwardRef } from "react";
+import useIsDeprecatedForm from "../../hooks/internal/useIsDeprecatedForm";
+import { ComponentWithStaticProperties } from "../../types";
+import { ComponentNames } from "../componentNames";
+import { defaultCheckboxGroupProps } from "./defaultCheckboxGroupProps";
+import { default as CheckboxGroupInternal } from "./CheckboxGroup";
+import { default as CheckboxGroupDeprecated } from "./CheckboxGroup_deprecated";
```
Comment (alifa20) [inline src/components/CheckboxGroup/index.tsx:7]: might be better to disable default same as LSC :D

File: src/components/Field/Field.tsx [hunk 9]
```diff
@@ -1,89 +1,12 @@
...
-        Record<string, any>,
-        ValidationError
-      >;
-      defaultValue?: FieldPathValue<TFieldValues, Name>;
-      // can't use Component<Props> because typescript doesn't allow dynamic rendering
-      // of components with arbitrary properties and refs
-      as: ComponentType<Props>;
-    } & Props
-  : "Component in `as=` expects a different value than the one provided by `name=`";
+import { FieldComponent, ValidProps, ValueProps } from "./types";
```
Comment (alifa20) [inline src/components/Field/Field.tsx:5]: 👍

File: src/hooks/useBasisForm/validation/validateCheckboxGroup.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import {
+  CheckboxOptions,
+  InternalCheckboxGroupProps,
+} from "../../../components/CheckboxGroup/types";
```
Comment (alifa20) [inline src/hooks/useBasisForm/validation/validateCheckboxGroup.ts:4]: Do you have relative imports setup ? too many `../...`

File: yarn.lock [hunk 11]
```diff
@@ -2727,7 +2727,7 @@ chalk@^3.0.0:
     ansi-styles "^4.1.0"
     supports-color "^7.1.0"
 
-chalk@^4.0.0, chalk@^4.1.0:
+chalk@^4.0.0, chalk@^4.1.0, chalk@^4.1.1:
```
Comment (alifa20) [inline yarn.lock:2730]: let's revert these changes and run yarn install again. I think you have these because of not using exact version numbers

File: scripts/bumpversion.mjs [hunk 12]
```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env zx
```
Comment (alifa20) [inline scripts/bumpversion.mjs:1]: maybe put a reference comment to `zx` somewhere here.

File: src/components/CheckboxGroup/CheckboxGroup_deprecated.js [hunk 13]
```diff
@@ -124,6 +123,7 @@ function CheckboxGroup(props) {
           <InternalCheckbox
             name={`${name}.${key}`}
             parentName={name}
+            // eslint-disable-next-line react/prop-types
```
Comment (alifa20) [inline src/components/CheckboxGroup/CheckboxGroup_deprecated.js:126]: sigh

File: src/components/CheckboxGroup/types.ts [hunk 14]
```diff
@@ -0,0 +1,68 @@
+import React from "react";
+
+type StringOption = {
+  key: string;
+  label: string;
+};
+
+type NodeOption = {
+  key: string;
+  label: React.ReactNode;
```
Comment (alifa20) [inline src/components/CheckboxGroup/types.ts:10]: I suggest running react codemod and cleanup all import reacts since basis is on v17
```
https://github.com/reactjs/react-codemod#update-react-imports
```

File: src/components/Stack.tsx [hunk 15]
```diff
@@ -48,6 +52,7 @@ function Stack(props) {
   const flexWrapperCSS = useResponsivePropsCSS(props, DEFAULT_PROPS, {
     margin: responsiveMargin,
     width: responsiveSize("width"),
+    // @ts-ignore
```
Comment (alifa20) [inline src/components/Stack.tsx:55]: 😞

**Review by alifa20 — APPROVED** [review]

**Review by alifa20 — APPROVED** [review]
Just the yarn install , and some minor suggesitons

File: src/components/CheckboxGroup/index.tsx (hunk 8 above)
Comment (zecuria) [reply src/components/CheckboxGroup/index.tsx:7]: That's what misha has throughout basis so won't do it until he comes back

File: src/hooks/useBasisForm/validation/validateCheckboxGroup.ts (hunk 10 above)
Comment (zecuria) [reply src/hooks/useBasisForm/validation/validateCheckboxGroup.ts:4]: Basis does not have absolute imports, I would have to run the change past misha before making that
